clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -682,7 +696,7 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                 shard->last_signal_context_ = { 0, {}, false };
                 // We have not seen any instr in the outermost scope that we just
                 // discovered.
-                shard->last_instr_in_cur_context_ = {};
+                shard->last_instr_in_cur_context_.instr = {};
```

Comment:
But if the "has_decoded" (see comment about renaming) is what determines whether this is valid shouldn't we clear that?